recipes/ogre/1.x/conanfile.py

```diff
@@ -0,0 +1,293 @@
+        "ogre_glsupport_use_egl": True,
+        "OpenEXR_Half_LIBRARY": "OpenEXR_Half_LIBRARY-NOTFOUND",
+        "OpenEXR_Half_LIBRARY_DEBUG": "OpenEXR_Half_LIBRARY_DEBUG-NOTFOUND",
+        "OpenEXR_INCLUDE_DIR": "OpenEXR_INCLUDE_DIR-NOTFOUND",
+        "OpenEXR_Iex_LIBRARY": "OpenEXR_Iex_LIBRARY-NOTFOUND",
+        "OpenEXR_Iex_LIBRARY_DEBUG": "OpenEXR_Iex_LIBRARY_DEBUG-NOTFOUND",
+        "OpenEXR_IlmImf_LIBRARY": "OpenEXR_IlmImf_LIBRARY-NOTFOUND",
+        "OpenEXR_IlmImf_LIBRARY_DEBUG": "OpenEXR_IlmImf_LIBRARY_DEBUG-NOTFOUND",
+        "OpenEXR_IlmThread_LIBRARY": "OpenEXR_IlmThread_LIBRARY-NOTFOUND",
+        "OpenEXR_IlmThread_LIBRARY_DEBUG": "OpenEXR_IlmThread_LIBRARY_DEBUG-NOTFOUND",
+        "Qt5_DIR": "Qt5_DIR-NOTFOUND",
+        "SDL2_DIR": "${ogre-build-dir}/Dependencies/cmake",
+        "pugixml_DIR": "${ogre-build-dir}/Dependencies/lib/cmake/pugixml"
+    }
+    exports_sources = "CMakeLists.txt", "patches/**"
+    _cmake = None
+
+    def requirements(self):
+        self.requires("cppunit/1.15.1")
+        self.requires("freeimage/3.18.0")
+        self.requires("boost/1.75.0")
+        self.requires("freetype/2.11.1")
+        self.requires("openexr/2.5.7")
+        self.requires("poco/1.11.2")
+        self.requires("tbb/2020.3")
+        self.requires("zlib/1.2.12")
+        self.requires("zziplib/0.13.71")
+        self.requires("openssl/1.1.1o", override=True)
+        self.requires("xorg/system")
+        if self.options.ogre_glsupport_use_egl:
+            self.requires("egl/system")
+        else:
+            self.requires("libglvnd/1.4.0")
+
+
+    def validate(self):
+        """
+         OGRE 1.x is very old and will not work with latest gcc, clang and msvc compielrs
+         TODO: determine incompatible msvc compilers
+        """
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) >= 11:
+            raise ConanInvalidConfiguration("OGRE 1.x not supported with gcc 11")
+        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) >= 13:
+            raise ConanInvalidConfiguration("OGRE 1.x not supported with clang 13")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake is not None:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["OGRE_STATIC"] = not self.options.shared
+        self._cmake.definitions["OGRE_CONFIG_DOUBLE"] = self.options.ogre_set_double
+        self._cmake.definitions["OGRE_CONFIG_NODE_INHERIT_TRANSFORM"] = False
+        self._cmake.definitions["OGRE_GLSUPPORT_USE_EGL"] = self.options.ogre_glsupport_use_egl
+        self._cmake.configure()
+        return self._cmake
```

Comment:
```suggestion
    @functools.lru_cache(1)
    def _configure_cmake(self):
        cmake = CMake(self)
        cmake.definitions["OGRE_STATIC"] = not self.options.shared
        cmake.definitions["OGRE_CONFIG_DOUBLE"] = self.options.ogre_set_double
        cmake.definitions["OGRE_CONFIG_NODE_INHERIT_TRANSFORM"] = False
        cmake.definitions["OGRE_GLSUPPORT_USE_EGL"] = self.options.ogre_glsupport_use_egl
        if not tools.valid_min_cppstd(self, 11):
            cmake.definitions["CMAKE_CXX_STANDARD"] = 11 # for OpenEXR
        cmake.configure()
        return cmake
```

Reply:
@ericLemanissier  thnx for the suggstions. Can you please also explain why `@functiontools.lru_cache(1)` is needed here ?

Reply:
it make python cache the result of the function for future calls, so that we don't have to use a dedicated member variable self._cmake